pkg/provider/kubernetes/gateway/kubernetes.go

```diff
@@ -927,9 +1209,127 @@ func loadServices(client Client, namespace string, targets []v1alpha1.HTTPRouteF
 	return wrrSvc, services, nil
 }
 
-func getProtocol(portSpec corev1.ServicePort, portName string) string {
+// loadTCPServices is generating a WRR service, even when there is only one target.
```

Comment:
that is a lot of common code with loadServices. might be worth refactoring into one func with a boolean argument or something. but not a blocker of course.